src/build/arg/arg_value.rs

```diff
@@ -61,6 +62,24 @@ impl<'help> ArgValue<'help> {
             Some(self.name)
         }
     }
+
+    /// TODO
+    pub fn get_name_and_aliases(&self) -> Vec<&str> {
+        [self.name].iter().chain(&self.aliases).copied().collect()
```

Comment:
`matches` calls this for every `ArgValue` it evaluates.  Allocating a `Vec` each time is a bit pricey.

Can we make the return type `impl Iterator<Item=&str>` and skip the `.collect()` call?

Reply:
I would prefer that as well, but I don't know how to append the name then, because we cannot return a reference to a temporary value.
![grafik](https://user-images.githubusercontent.com/11978847/134959529-5a7e489a-0e46-4fca-810c-b8d021b6a359.png)

Reply:
Try
```rust
IntoIterator::into_iter([self.name]).chain(&self.aliases).copied()
```

I'm assuming the temporary is the array and using `into_iterator` will move it.

Reply:
my bad there is a better way:
```rs
    pub fn get_name_and_aliases(&self) -> impl Iterator<Item = &str> {
        iter::once(&self.name).chain(&self.aliases).copied()
    }
```

Reply:
I always forgot about that. I either use `[]` or `Some()`